src/Orleans.Runtime/Catalog/ActivationCollector.cs

```diff
@@ -452,16 +452,7 @@ private async Task CollectActivationsImpl(bool scanStale, TimeSpan ageLimit, Can
             var number = Interlocked.Increment(ref collectionNumber);
             long memBefore = GC.GetTotalMemory(false) / (1024 * 1024);
 
-            if (logger.IsEnabled(LogLevel.Debug))
-            {
-                logger.LogDebug(
-                    (int)ErrorCode.Catalog_BeforeCollection,
-                    "Before collection #{CollectionNumber}: memory: {MemoryBefore}MB, #activations: {ActivationCount}, collector: {CollectorStatus}",
-                    number,
-                    memBefore,
-                    _activationCount,
-                    ToString());
-            }
+            LogBeforeCollection(number, memBefore, _activationCount, this);
 
             List<ICollectibleGrainContext> list = scanStale ? ScanStale() : ScanAll(ageLimit);
             CatalogInstruments.ActivationCollections.Add(1);
```

Comment:
I'm not sure if it's possible to convert the log
```cs
if (logger.IsEnabled(LogLevel.Debug)) logger.LogDebug("CollectActivations {Activations}", list.ToStrings(d => d.GrainId.ToString() + d.ActivationId));
```
to the `LoggerMessage` syntax, is it possible to have custom formatters for parameters? (to call the `.ToStrings` method conditionally like you're doing now)

Reply:
@ReubenBond I wonder if we could introduce a utility struct to simplify this
```cs
private struct ActivationsLogValue(List<ICollectibleGrainContext> list)
{
    public override string ToString() => list.ToStrings(d => d.GrainId.ToString() + d.ActivationId);
}
```
and then have the logging method have an argument of type `ActivationsLogValue`

Reply:
@ReubenBond thoughts on this?